conda/exceptions.py

```diff
@@ -328,8 +328,17 @@ def __init__(self, url, target_full_path, expected_md5sum, actual_md5sum):
 
 
 class PackageNotFoundError(CondaError):
-    def __init__(self, message, **kwargs):
-        super(PackageNotFoundError, self).__init__(message, **kwargs)
+
+    def __init__(self, message, channel_urls=(), **kwargs):
+        from .resolve import dashlist
+        channels = dashlist(channel_urls)
+        msg = dals("""
+        %(text)s
+
+        We have searched for the package in the following channels:
+        %(channels)s
+        """)
+        super(PackageNotFoundError, self).__init__(msg, text=message[0], channels=channels)
```

Comment:
I'm temporilly commenting out test_PackageNotFoundError because of `text=message[0]`.
I will bring back the test when I sort out the PackageNotFoundError and NoPackageFoundError.